netbox/dcim/filtersets.py

```diff
@@ -1511,22 +1511,16 @@ class Meta:
         ]
 
     def filter_device(self, queryset, name, value):
-        try:
-            devices = Device.objects.filter(**{'{}__in'.format(name): value})
-            vc_interface_ids = []
-            for device in devices:
-                vc_interface_ids.extend(device.vc_interfaces().values_list('id', flat=True))
-            return queryset.filter(pk__in=vc_interface_ids)
-        except Device.DoesNotExist:
-            return queryset.none()
-
-    def filter_device_id(self, queryset, name, id_list):
         # Include interfaces belonging to peer virtual chassis members
         vc_interface_ids = []
         try:
-            devices = Device.objects.filter(pk__in=id_list)
+            devices = Device.objects.filter(**{'{}__in'.format(name): value})
             for device in devices:
-                vc_interface_ids += device.vc_interfaces(if_master=False).values_list('id', flat=True)
+                # Hack to show all VC member interfaces when requested
+                if self.request is not None and 'vc_interfaces' in self.request.GET.keys():
```

Comment:
We can't support this pattern. Any separate query parameter needs to have its own filter defined on the filter set for validation and documentation. We'll need to figure out a different mechanism to toggle the inclusion of VC interfaces. Defining an entirely separate filter, while perhaps not ideal, is one option.

Reply:
Fair enough,

I think it might be best to consider this a breaking change, rebase to feature and then remove the custom logic for both.  Instead device returns only specific devices.

With that, introduce a new filter, `virtual_chassis` that will return all the interfaces as part of a virtual chassis.